ui/libs/pages/search/src/lib/components/filter/filter.component.ts

```diff
@@ -0,0 +1,116 @@
+import {Component, EventEmitter, Input, Output, ViewChild} from '@angular/core';
+import { addFq, IFilter, removeFq } from '@eosc-search-service/search';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FlatNode } from '@eosc-search-service/layout';
+import { FormControl } from '@angular/forms';
+import { debounceTime } from 'rxjs';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import {assertNotNull} from "@eosc-search-service/types";
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-filter',
+  template: `
+    <div class="filter" *ngIf="filter">
+      <span class="filter-title"
+        ><b>{{ filter.title }}</b></span
+      >
+      <input
+        *ngIf="filter.showMore"
+        [attr.placeholder]="'Search in ' + filter.title"
+        class="query-input form-control form-control-sm"
+        [formControl]="queryFc"
+      />
+      <div [ngClass]="{filter__viewport: filter.showMore}" (scroll)="onScroll($event)" #content>
+        <ess-checkboxes-tree
+          [data]="filter.data"
+          (checkboxesChange)="addFilter($event)"
+        ></ess-checkboxes-tree>
+      </div>
+      <a
+        *ngIf="isShowMoreVisible || filter.showMore"
+        class="show-more"
+        href="javascript:undefined"
+        (click)="toggleShowMore.emit()"
+        >
+        <ng-container *ngIf="!filter.showMore; else showLessRef" i18n>Show more</ng-container>
+        <ng-template #showLessRef><ng-container i18n>Show less</ng-container></ng-template>
+      </a
+      >
+    </div>
+  `,
+  styles: [
+    `
+      .filter__viewport {
+        max-height: 400px;
+        overflow: auto;
+      }
+
+      .query-input {
+        margin-bottom: 5px;
+      }
+
+      .filter {
+        margin-bottom: 10px;
+      }
+      .filter-title {
+        padding-bottom: 6px;
+        display: block;
+      }
+      .ant-tree {
+        background: none !important;
+      }
+
+      .show-more {
+        font-size: 11px;
+      }
+    `,
+  ],
+})
+export class FilterComponent {
+  @ViewChild('content', {static: false}) content?: any;
+  @Input() filter: IFilter | null = null;
+  @Output() toggleShowMore = new EventEmitter<void>();
+  // @Output() addFilter = new EventEmitter
+  @Output() searchFilter = new EventEmitter<string>();
+  @Output() fetchMore = new EventEmitter<string>();
+  get isShowMoreVisible(): boolean {
```

Comment:
Using `get` in HTML template is a terrible idea. It will recalculate the results a lot. Please use Rxjs or ngOnChanges (?).